odoo/tests/common.py

```diff
@@ -284,6 +286,26 @@ def _tearDownPreviousClass(self, test, result):
                                                                         className,
                                                                         info=exc)
 
+class OdooSuite(BackportSuite):
+    def _handleClassSetUp(self, test, result):
+        if not hasattr(result, 'stats') \
+                or not stats_logger.isEnabledFor(logging.INFO):
+            super()._handleClassSetUp(test, result)
+            return
+
+        with result.collectStats(type(test), 'setUpClass'):
```

Comment:
`id` should probably be passed here entirely, that would make the collector not reliant on generating the id itself, which would be a better idea.